### PR TITLE
ansible: dynamically pull vendor files from the secrets repo

### PR DESCRIPTION
**Pull request by gdams**
This PR allows us to pull all vendor file secrets from https://github.com/AdoptOpenJDK/secrets/tree/master/vendor_files

## Requirements

- The user running the job has to have permission to clone the secrets repo
- The user must have gpg access to the secrets. (see more information in the secrets repo readme)
- The user must have dotgpg installed to extract the secrets

### CODE REVIEW

**Review by aahlenst — COMMENTED** [review]
LGTM.

**Review by aahlenst — APPROVED** [review]
Reviewing again because I'm not clever enough to pick the correct radio button 🙄

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Get_Vendor_Files/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+##############
+# VENDOR_FILES
+##############
+
+- name: check out AdoptOpenJDK/secrets
+  local_action: git repo=git@github.com:AdoptOpenJDK/secrets.git dest=vendor_files
+  register: git_clone
+
+- name: check if dotgpg is installed
+  local_action: command command -v dotgpg
```
Comment (Willsparker) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Get_Vendor_Files/tasks/main.yml:11]: Is there meant to be 2 `commands` there? The output is the same, regardless.

Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Get_Vendor_Files/tasks/main.yml:11]: Yup, the first command is an ansible plugin, the second is actually running command -v

Comment (Willsparker) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Get_Vendor_Files/tasks/main.yml:11]: oh ignore me, the first command is to tell `local_action` what it's doing, my bad

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Get_Vendor_Files/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+##############
+# VENDOR_FILES
+##############
+
+- name: check out AdoptOpenJDK/secrets
+  local_action: git repo=git@github.com:AdoptOpenJDK/secrets.git dest=vendor_files
+  register: git_clone
```
Comment (Willsparker) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Get_Vendor_Files/tasks/main.yml:8]: It doesn't look like you use the `git_clone` var?

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Get_Vendor_Files/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+---
+##############
+# VENDOR_FILES
+##############
+
+- name: check out AdoptOpenJDK/secrets
+  local_action: git repo=git@github.com:AdoptOpenJDK/secrets.git dest=vendor_files
+  register: git_clone
```
Comment (Willsparker) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Get_Vendor_Files/tasks/main.yml:8]: Same here.

**Review by Willsparker — CHANGES_REQUESTED** [review]

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Get_Vendor_Files/tasks/main.yml (hunk 3 above)
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Get_Vendor_Files/tasks/main.yml:8]: fixes

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Get_Vendor_Files/tasks/main.yml (hunk 2 above)
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Get_Vendor_Files/tasks/main.yml:8]: fixed

**Comment by Willsparker** [comment]
Thanks for those changes- So I'm trying to run this locally and I'm getting this error message (I'm running the playbook on a Ubuntu1604 Vagrant VM, connecting via the ssh port forward, so the host's file just contains '127.0.0.1:2222'
```
TASK [Get_Vendor_Files : check out AdoptOpenJDK/secrets] *********************************************************************************************************************************************
fatal: [127.0.0.1]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: Warning: Permanently added '[localhost]:2222' (ECDSA) to the list of known hosts.\r\nwill@localhost: Permission denied (publickey,password).", "unreachable": true}
```
This is almost definitely an issue with how I've set it up, as opposed to the playbook, but I was wondering if you could shed some light?

**Comment by gdams** [comment]
yeah so you'll need to have dotpg and you github ssh key installed on that machine in order for it to work

**Comment by Willsparker** [comment]
Hmm, it's an issue with the way `local_action` is working:
```
TASK [Get_Vendor_Files : check out AdoptOpenJDK/secrets] *********************************************************************************************************************************************
task path: /home/will/Documents/vendor_files/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Get_Vendor_Files/tasks/main.yml:6
<localhost> ESTABLISH SSH CONNECTION FOR USER: will
<localhost> SSH: EXEC ssh -F /dev/null -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=2222 -o 'IdentityFile="/home/will/Documents/vendor_files/ansible/id_rsa"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="will"' -o ConnectTimeout=30 -o ControlPath=/home/will/.ansible/cp/0c5ff6558d localhost '/bin/sh -c '"'"'echo ~vagrant && sleep 0'"'"''
<localhost> (255, '', 'will@localhost: Permission denied (publickey,password).\r\n')
```
So it's connecting to `will@localhost` which is fine, except it's trying to connect to port `2222` which is currently assigned to the VM's ssh port.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Get_Vendor_Files/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+---
+##############
+# VENDOR_FILES
+##############
+
+- name: check out AdoptOpenJDK/secrets
+  local_action: git repo=git@github.com:AdoptOpenJDK/secrets.git dest=vendor_files
+
+- name: check if dotgpg is installed
+  local_action: command command -v dotgpg
```
Comment (Willsparker) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Get_Vendor_Files/tasks/main.yml:10]: Can we change these to `shell command -v dotgpg`. For whatever reason, the `command` builtin module couldn't find `dotgpg` installed on my laptop, but the `shell` module could.

**Review by Willsparker — CHANGES_REQUESTED** [review]

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Get_Vendor_Files/tasks/main.yml (hunk 4 above)
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Get_Vendor_Files/tasks/main.yml]: done

**Review by Willsparker — APPROVED** [review]
LGTM :-)

**Comment by gdams** [comment]
@sxa can you confirm that this change also works for your setup? You're the other main person that regularly uses vendor_files to setup machines

**Comment by karianna** [comment]
Same linter failure as the other PR...

**Comment by gdams** [comment]
Merging as no objections from @sxa for 4 days. We can always back this change out if there is a problem

**Comment by sxa** [comment]
@gdams I think you may have done this while I was on vacation. This has caused a problem for the IBM team also using the playbooks. Pulling from the secrets repo absolutely shouldn't be the default operation - others should be able to run the playbooks as far as possible.

**Comment by gdams** [comment]
My understanding was that IBM skipped the AdoptOpenJDK tag??

**Pull request merged**